lib/matplotlib/_docstring.py

```diff
@@ -54,8 +54,8 @@ def __missing__(self, key):
         name = key[:-len(":kwdoc")]
         from matplotlib.artist import Artist, kwdoc
         try:
-            cls, = [cls for cls in _api.recursive_subclasses(Artist)
-                    if cls.__name__ == name]
+            cls, = (cls for cls in _api.recursive_subclasses(Artist)
```

Comment:
I would leave these cases as lists, as it is exactly as long to type, and unpacking from a list is faster than doing so from a generator:
```
In [4]: %timeit x, = (x for x in range(3) if x % 2 == 1)
533 ns ± 1.99 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [5]: %timeit x, = [x for x in range(3) if x % 2 == 1]
487 ns ± 1.91 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [6]: %timeit x, = (x for x in range(1))
418 ns ± 0.467 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [7]: %timeit x, = [x for x in range(1)]
357 ns ± 1.08 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```
(there are other similar examples elsewhere).

(In expressions like `"".join(x for x in ...)` it's not really worth bothering adding an extra pair of brackets so leaving the genexp is fine.)

Reply:
I was also surprised that these were rewritten as my earlier investigations showed no obvious benefit for tuples and comprehensions. Will revert in my other PR.

Reply:
:+1: Sure, I reverted these. I don't have a major preference either way.